src/components/Toolbar.tsx

```diff
@@ -31,18 +31,9 @@ const ARROW_SCROLL_BUFFER = 20
 // eslint-disable-next-line jsdoc/require-jsdoc
 const mapStateToProps = (state: State) => {
 
-  const { cursor, fontSize, thoughts, isLoading, toolbarOverlay, scrollPrioritized, showTopControls } = state
-
   return {
     dark: theme(state) !== 'Light',
-    isLoading,
-    fontSize,
-    scrollPrioritized,
-    toolbarOverlay,
-    // We cannot know if any one the shortcut's active status,has changed, so we re-render everytime thoughts or cursor is changed
-    thoughts,
-    cursor,
-    showTopControls
+    ...state
```

Comment:
Re-rendering the Toolbar when anything in the state changes is overkill and likely to introduce performance issues. It should only re-render when it needs to.